lua/terrortown/entities/items/item_ttt_radar/cl_init.lua

```diff
@@ -271,32 +251,42 @@ end
 net.Receive("TTT_CorpseCall", TTT_CorpseCall)
 
 local function ReceiveRadarScan()
-	local num_targets = net.ReadUInt(8)
+	local num_targets = net.ReadUInt(16)
 
 	RADAR.targets = {}
 
 	for i = 1, num_targets do
-		local sr = net.ReadUInt(ROLE_BITS)
+		local pos = Vector(
+			net.ReadInt(32),
+			net.ReadInt(32),
+			net.ReadInt(32)
+		)
 
-		local pos = Vector()
-		pos.x = net.ReadInt(32)
-		pos.y = net.ReadInt(32)
-		pos.z = net.ReadInt(32)
+		local hasSubrole = net.ReadBool()
```

Comment:
You could use a free role id too. Ref.: https://github.com/TTT-2/TTT2/blob/e131d95c08e7114e2d08515b2380999f0decc628/lua/includes/modules/roles.lua#L287

Reply:
I could do this. However for the readability sake I don't think it is nice to use a "random" number

Reply:
You are right, but regarding the upcoming change with `ROLE_NONE` or `ROLE_UNKNOWN` (...) this would help to remember :D

Reply:
Yes, I added a boolean because someone said that the unknown role thing would be a secondary state, not a "fake role". I think it was saibotk. I'm not sure though.

Reply:
Okay then feel free to resolve if it's clear :)

Reply:
I'll add a remark to this file in the trello ticket about the unknown role state.